helm/configurations/values-dev.yaml

```diff
@@ -1,4 +1,7 @@
 # REANA components pointing to `latest`, locally built master branch
+debug:
+  enabled: true
+
```

Comment:
This would not allow to test production-like deployment before release builds, isn't it? Things can behave differently when code is mounted, so it would be good to retain that option. I mean we have been distinguishing three "local" modes:

- production, for people testing out REANA, using tagged docker images from docker hub
- prod-like development, using `latest` images, but without code mounting
- debug-like development, using `latest` images, with code mounting

The second option allows to run CI tests before releasing images for example. (Or were you thinking of switching to  tagged builds for that mode?) 

We could solve this by introducing several `values.yaml` files:

- production, default `helm/reana/values.yaml`, interesting for people testing out REANA
- prod-like development, using `helm/configurations/kind-latest.yaml`
- debug-like development with live code mounts, using `helm/configurations/kind-latest-debug.yaml`

This would allow to have in the future more configurations, say `kind-latest-multi-node.yaml` and whatnot, or `-vc3.yaml` and whatnot...

The `reana-dev cluster-deploy` would then optionally take which configuration to deploy.

P.S. Can use "local-" instead of leading "kind-..." or other such suitable name...

Recall also old musings https://github.com/reanahub/reana/pull/289#discussion_r409441312

Reply:
Since `debug`/`dev` is a very clear and recurrent thing to do, I have added a `--debug` flag, just as in `reana-dev cluster-create` where `--debug` will mount the code transparently. This flag would mean in this context to simply add the `debug.enabled=true` to Helm install as a `--set key=value` option:
> production, for people testing out REANA, using tagged docker images from docker hub
1. Would be installing latest stable version from [our Helm repo](https://reanahub.github.io/reana/) because it wouldn't be safe to use the `values.yaml` as many times we update the templates but not the docker images, so we would break this mode.
> prod-like development, using latest images, but without code mounting
2. Running `reana-dev cluster-deploy`
> debug-like development, using latest images, with code mounting
3. Running `reana-dev cluster-deploy --debug`
> We could solve this by introducing several values.yaml files:
4. Additionally to the "basic" setup, we could add a `--file` or `--values` option so we have room for growing towards `*-vc3.yaml` etc....

Would it be of interest for this tool to encapsulate (1) as well?

Reply:
> Would it be of interest for this tool to encapsulate (1) as well?

We can leave it out indeed, you are right about template vs docker image updates... We could have another (simpler, non-reana-dev) CI sequence to test latest releases from the end user point of view. In this way `reana-dev` will remain for local developments only